src/main/java/org/wso2/carbon/mongodb/user/store/mgt/MongoDBUserStoreManager.java

```diff
@@ -0,0 +1,2559 @@
+        }
+        if (dataSource == null) {
+            throw new UserStoreException("User management data source is null");
+        }
+        properties.put(UserCoreConstants.DATA_SOURCE, dataSource);
+        realmConfig.setUserStoreProperties(MongoDBRealmUtil.getMongoProperties(realmConfig.getUserStoreProperties()));
+
+        this.persistDomain();
+        doInitialSetup();
+        if (!skipInitData && realmConfig.isPrimary()) {
+            addInitialAdminData(Boolean.parseBoolean(realmConfig.getAddAdmin()), !isInitSetupDone());
+        }
+        // Initialize user roles cache
+        initUserRolesCache();
+    }
+
+    public static void setDBDataSource(DataSource source) {
+        dataSourceLocal = source;
+    }
+
+    /**
+     * Get all user properties belong to provided user profile
+     *
+     * @param userName      username of user
+     * @param propertyNames names of properties to get
+     * @param profileName   profile name of user
+     * @return map object of properties
+     */
+    protected Map<String, String> getUserPropertyValues(String userName, String[] propertyNames, String profileName)
+            throws UserStoreException {
+
+        if (profileName == null) {
+            profileName = UserCoreConstants.DEFAULT_PROFILE;
+        }
+        MongoPreparedStatement prepStmt = null;
+        String[] propertyNamesSorted = propertyNames.clone();
+        Arrays.sort(propertyNamesSorted);
+        Map<String, String> map = new HashMap<>();
+
+        try {
+            if (db == null) {
+                db = loadUserStoreSpecificDataSource();
+            }
+            String mongoQuery = realmConfig.getUserStoreProperty(MongoDBRealmConstants.GET_PROPS_FOR_PROFILE);
+            prepStmt = new MongoPreparedStatementImpl(db, mongoQuery);
+            prepStmt.setString("users.UM_USER_NAME", userName);
+            prepStmt.setString("UM_PROFILE_NAME", profileName);
+
+            if (mongoQuery.contains(UserCoreConstants.UM_TENANT_COLUMN)) {
+                prepStmt.setInt("users.UM_TENANT_ID", tenantId);
+                prepStmt.setInt("UM_TENANT_ID", tenantId);
+            }
+            //noinspection deprecation
+            AggregationOutput aggregationOutput = prepStmt.aggregate();
+            Iterable<DBObject> results = aggregationOutput.results();
+            for (DBObject object : results) {
+                object.removeField("_id");
+                object.removeField("users");
+                Set<String> keys = object.keySet();
+                for (String key : keys) {
+                    String value = object.get(key).toString();
+                    if (Arrays.binarySearch(propertyNamesSorted, key) < 0) {
+                        continue;
+                    }
+                    map.put(key, value);
+                }
+            }
+        } catch (UserStoreException e) {
+            throw new UserStoreException("Failed to get the user property values", e);
```

Comment:
Why are we catching and throwing the same exception again?

Reply:
Removed the unnecessary catch parts, which were capturing and throwing same exceptions.